src/root/views/GlobalThreeW/Charts/index.tsx

```diff
@@ -0,0 +1,174 @@
+import React from 'react';
+import {
+  Cell,
+  Bar,
+  BarChart,
+  Pie,
+  PieChart,
+  XAxis,
+  YAxis,
+  Tooltip,
+  Legend,
+  ResponsiveContainer,
+} from 'recharts';
+
+import Card from '#components/Card';
+
+import styles from './styles.module.scss';
+
+interface ChartData {
+  key: number,
+  name: string,
+  value: number,
+}
+
+const PIE_COLORS = ['#f64752', '#fa999f', '#f87079'];
+
+
+interface PieChartProps {
+  data: any[],
+  heading: string,
+  className?: string;
+}
+function GoPieChart(props: PieChartProps) {
+  const {
+    data,
+    className,
+  } = props;
+
+  const chartData: ChartData[] = React.useMemo(() => {
+    if (data) {
+      const dataArr = data.map((datum: any) => {
+        return {
+          key: datum["programme_type"],
+          name: datum["programme_type_display"],
+          value: datum["count"],
+        };
+      });
+      return dataArr;
+    }
+
+    return [];
+  }, [data]);
+
+  return (
+    <Card
+      className={className}
+      title={props.heading}
+    >
+      <ResponsiveContainer
+        className={styles.chartContainer}
+      >
+        <PieChart>
+          <Pie
+            data={chartData}
+            dataKey="value"
+          >
+            { data.map((entry, index) => (
+              <Cell
+                key={`cell-${index}`}
+                fill={PIE_COLORS[index % PIE_COLORS.length]}
+              />
+            ))}
+          </Pie>
+          <Legend
+            iconType="circle"
+            verticalAlign="middle"
+            align="right"
+            layout="vertical"
+          />
+        </PieChart>
+      </ResponsiveContainer>
+    </Card>
+  );
+}
+
+function CustomYAxisTick (props: {
+  y: number;
+  payload: unknown;
+}) {
+  const { y, payload } = props;
+  const { value } = payload as { value: number };
+
+  return (
+    <g transform={`translate(${0},${y})`}>
+      <text
+        x={0}
+        y={0}
+        textAnchor="start"
+        className={styles.tick}
+      >
+        {value}
+      </text>
+    </g>
+  );
+}
+
+interface BarChartProps {
+  data: any[],
+  heading: string,
+  className?: string;
+}
+
+function GoBarChart(props: BarChartProps) {
+  const {
+    data,
+    className,
+  } = props;
+
+  const chartData = React.useMemo(() => {
+    if (data && data.length > 0) {
+      const keys = Object.keys(data[0]);
+      const dataArr = data.map((datum: any) => {
+        return { key: datum[keys[0]], name: datum[keys[1]], value: datum["count"] };
+      });
+
+      return dataArr
+        .sort((a, b) => b.value - a.value)
+        .splice(0, 5); // Only show top 5
```

Comment:
Shouldn't this be configurable?

Reply:
This is only for the Global 3W page, so I guess it's okay